tests/Microsoft.Identity.Test.Unit/PublicApiTests/ExtensiblityTests.cs

```diff
@@ -218,6 +218,121 @@ public async Task ValidateAppTokenProviderAsync()
             }
         }
 
+        [TestMethod]
+        public async Task ValidateAdditionalCacheParametersAreStored()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                              .WithAuthority("https://login.microsoftonline.com/tid/")
+                              .WithClientSecret(TestConstants.ClientSecret)
+                              .WithHttpManager(httpManager)
+                              .BuildConcrete();
+
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+                var result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                    .WithAdditionalCacheParameters(new List<string> { "additional_param1", "additional_param2", "additional_param3" })
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                var parameters = app.AppTokenCacheInternal.Accessor.GetAllAccessTokens().Single().AdditionalCacheParameters;
+                Assert.IsTrue(parameters.Count == 3);
+
+                parameters.TryGetValue("additional_param1", out string additionalParam1);
+                parameters.TryGetValue("additional_param2", out string additionalParam2);
+                parameters.TryGetValue("additional_param3", out string additionalParam3);
+
+                Assert.AreEqual("value1", additionalParam1);
+                Assert.AreEqual("value2", additionalParam2);
+                Assert.AreEqual("value3", additionalParam3);
+
+                Assert.AreEqual("Bearer", result.TokenType);
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+                //Verify cache parameters still exist
+                result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                    .WithAdditionalCacheParameters(new List<string> { "additional_param1", "additional_param2", "additional_param3" })
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual("Bearer", result.TokenType);
+                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+
+                parameters = app.AppTokenCacheInternal.Accessor.GetAllAccessTokens().Single().AdditionalCacheParameters;
```

Comment:
`MsalAccessTokenCacheItem` is not exposed to app developers. The values are read via `AuthenticationResult.AdditionalResponseParameters` and the test should validate that.

Reply:
Im not sure this is the case, the goal here is to ensure that the access token caches the specific values in the additional params api, not what is in the auth result. The new api has no affect on the `AuthenticationResult.AdditionalResponseParameters`. All additional values will be present in `AuthenticationResult.AdditionalResponseParameters` even if we only specify some of them in the additional params api to be cached. Also, we already know the values in `AuthenticationResult.AdditionalResponseParameters` because they are hard coded. Testing that the same values are returned in the auth result would be testing some other part of the code that is not in this PR no?

Reply:
Also, `AuthenticationResult.AdditionalResponseParameters` is null when acquiring tokens from the cache

Reply:
Discussed offline, will update `AuthenticationResult.AdditionalResponseParameters` to reflect what is cached in the access token.